src/AutoRest.CSharp/LowLevel/Output/LowLevelClient.cs

```diff
@@ -121,33 +125,97 @@ public static IEnumerable<LowLevelClientMethod> BuildMethods(RestClientBuilder b
             }
         }
 
-        private IEnumerable<ConstructorSignature> BuildPublicConstructors()
+
+        private (ConstructorSignature[] PrimaryConstructors, ConstructorSignature[] SecondaryConstructors) BuildPublicConstructors()
         {
-            if (!_hasPublicConstructors)
+            if (!IsSubClient)
+            {
+                var requiredParameters = RestClientBuilder.GetRequiredParameters(Parameters).ToArray();
```

Comment:
`Parameters` property is irrelevant and misleading at this point - we don't have any constructor that has set of parameters that match `Parameters` collection, so we need to remove it completely. To determine `requiredParameters` and `optionalParameters`, we need to use builder directly, e.g.:
```cs
var parameters = builder.GetOrderedParametersByRequired();
var requiredParameters = RestClientBuilder.GetRequiredParameters(parameters);
var optionalParameters = RestClientBuilder.GetRequiredParameters(parameters).Append(CreateOptionsParameter());
```

Reply:
I don't want to calculate same parametes twice. How about adding an `orderedParameter` parameter to the method.

Reply:
Where do you need those for the second time?

Reply:
Ok, let's leave `Parameters` as they are right now. We can clean it up after GA.